reV/generation/generation.py

```diff
@@ -482,11 +489,85 @@ def time_index(self):
             elif step is not None:
                 time_index = time_index[::step]
 
-            self._time_index = self.handle_leap_ti(time_index,
-                                                   drop_leap=self._drop_leap)
+            time_index = self.handle_lifetime_index(time_index)
+            time_index = self.handle_leap_ti(time_index,
+                                             drop_leap=self._drop_leap)
+
+            self._time_index = time_index
 
         return self._time_index
 
+    def handle_lifetime_index(self, ti):
+        """Adjust the time index if modeling full system lifetime.
+
+        Parameters
+        ----------
+        ti : pandas.DatetimeIndex
+            Time-series datetime index with leap days.
+
+        Returns
+        -------
+        ti : pandas.DatetimeIndex
+            Time-series datetime index.
+        """
+        life_var = "system_use_lifetime_output"
+        lifetime_periods = []
+        for sam_meta in self.sam_metas.values():
+            if life_var in sam_meta and sam_meta[life_var] == 1:
+                lifetime_period = sam_meta["analysis_period"]
+                lifetime_periods.append(lifetime_period)
+            else:
+                lifetime_periods.append(1)
+
+        if any(ltp > 1 for ltp in lifetime_periods):
```

Comment:
Can you please invert this check to be
```suggestion
        if not any(ltp > 1 for ltp in lifetime_periods):
            return ti
```
Then all the code below it can come down one indent, which is preferable

Reply:
Done